sycl/CMakeLists.txt

```diff
@@ -95,34 +108,108 @@ add_custom_target(sycl-headers ALL
-
 # SYCL runtime library
 add_subdirectory(source)
 
 # SYCL toolchain builds all components: compiler, libraries, headers, etc.
+set(SYCL_TOOLCHAIN_DEPS
+  sycl
+  clang
+  clang-offload-wrapper
+  clang-offload-bundler
+  llc
+  llvm-as
+  llvm-ar
+  llvm-dis
+  llvm-no-spir-kernel
+  llvm-spirv
+  llvm-link
+  llvm-objcopy
+  opt)
 add_custom_target( sycl-toolchain
-  DEPENDS sycl
-          clang
-          clang-offload-wrapper
-          clang-offload-bundler
-          llc
-          llvm-as
-          llvm-ar
-          llvm-dis
-          llvm-no-spir-kernel
-          llvm-spirv
-          llvm-link
-          llvm-objcopy
-          opt
+  DEPENDS ${SYCL_TOOLCHAIN_DEPS}
   COMMENT "Building SYCL compiler toolchain..."
 )
 
 add_subdirectory( test )
 add_subdirectory( unittests )
 add_subdirectory( tools )
+
+# Configure SYCL toolchain installation targets
+add_custom_target(deploy-sycl-toolchain)
+
+set(SYCL_BUNDLE_DEPS ${SYCL_TOOLCHAIN_DEPS} opencl clang-resource-headers)
+
+foreach(comp ${SYCL_BUNDLE_DEPS})
+    add_custom_command(TARGET deploy-sycl-toolchain PRE_BUILD
+        COMMAND ${CMAKE_COMMAND}
+        ARGS -DCOMPONENT="${comp}" -P "${CMAKE_BINARY_DIR}/cmake_install.cmake")
+endforeach()
+
+set(LLVM_LIT_EXT "")
+if(WIN32)
+  set(LLVM_LIT_EXT ".py")
+endif()
+add_custom_command(TARGET deploy-sycl-toolchain PRE_BUILD
+  COMMAND ${CMAKE_COMMAND}
+  ARGS -E copy ${CMAKE_BINARY_DIR}/bin/llvm-lit${LLVM_LIT_EXT}
+  ${CMAKE_INSTALL_PREFIX}/bin/llvm-lit${LLVM_LIT_EXT})
+
+# For some reason clang-cl is being created on Linux. Remove it so users
+# are not confused.
+if (NOT WIN32)
+  add_custom_command(TARGET deploy-sycl-toolchain POST_BUILD
+    COMMAND ${CMAKE_COMMAND}
+    ARGS -E remove ${CMAKE_INSTALL_PREFIX}/bin/clang-cl)
+endif()
+
+if (SYCL_DEPLOY_OPENCL)
+  install(TARGETS OpenCL EXPORT SYCL
+      LIBRARY
+        DESTINATION lib
+        COMPONENT opencl
+      ARCHIVE
+        DESTINATION lib
+        COMPONENT opencl
+      RUNTIME
+        DESTINATION bin
+        COMPONENT opencl
+      INCLUDES
+      DESTINATION include
+        COMPONENT opencl)
+
+  # If OpenCL target is not built with SYCL, we need to copy libraries manually
+  if(OpenCL_LIBRARIES)
+    set (RESOLVED_LIBS "")
+    foreach (file ${OpenCL_LIBRARIES})
+      get_filename_component(resolved_lib "${file}" REALPATH)
+      list (APPEND RESOLVED_LIBS "${resolved_lib}")
+    endforeach()
+    install(FILES ${RESOLVED_LIBS} DESTINATION lib COMPONENT opencl)
+    if(WIN32)
+      string(REPLACE ".lib" ".dll" DLL_LIBS "${RESOLVED_LIBS}")
+      install(FILES ${DLL_LIBS} DESTINATION bin COMPONENT opencl)
+    else(WIN32)
+      if(RESOLVED_LIBS MATCHES "libOpenCL.so.1.0.0")
+        # Workaround for some OpenCL ICD packages
+        add_custom_command(TARGET deploy-sycl-toolchain POST_BUILD
+            COMMAND ${CMAKE_COMMAND}
+          ARGS -E rename ${CMAKE_INSTALL_PREFIX}/lib/libOpenCL.so.1.0.0
+          ${CMAKE_INSTALL_PREFIX}/lib/libOpenCL.so)
```

Comment:
I feel something is wrong with this WA.
I suggest removing it... 
I think we should start with deploying only latest Khronos ICD, if it's downloaded and build from sources by this script.
I'm not sure there is a reason to deploy anything else.